src/plugins/console/public/application/components/variables/variables_flyout.tsx

```diff
@@ -0,0 +1,170 @@
+  onClose: () => void;
+  onSaveVariables: (newVariables: DevToolsVariable[]) => void;
+  variables: [];
+}
+
+export interface DevToolsVariable {
+  id: string;
+  name: string;
+  value: string;
+}
+
+export const DevToolsVariablesFlyout = (props: DevToolsVariablesFlyoutProps) => {
+  const [variables, setVariables] = useState<DevToolsVariable[]>(props.variables);
+  const formId = useGeneratedHtmlId({ prefix: '__console' });
+
+  const addNewVariable = useCallback(() => {
+    setVariables((v) => [...v, utils.generateEmptyVariableField()]);
+  }, []);
+
+  const deleteVariable = useCallback(
+    (id: string) => {
+      const updatedVariables = utils.deleteVariable(variables, id);
+      setVariables(updatedVariables);
+    },
+    [variables]
+  );
+
+  const onSubmit = useCallback(
+    (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+      props.onSaveVariables(variables.filter(({ name, value }) => name.trim() && value));
+    },
+    [props, variables]
+  );
+
+  const onChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>, id: string) => {
+      const { name, value } = event.target;
+      const editedVariables = utils.editVariable(name, value, id, variables);
+      setVariables(editedVariables);
+    },
+    [variables]
+  );
+
+  const columns: Array<EuiBasicTableColumn<DevToolsVariable>> = [
+    {
+      field: 'name',
+      name: i18n.translate('console.variablesPage.variablesTable.columns.variableHeader', {
+        defaultMessage: 'Variable',
+      }),
+      render: (name, { id }) => {
+        const isInvalid = name && !name.match(/^[a-zA-Z0-9]+$/g);
+        return (
+          <EuiFieldText
+            data-test-subj="variablesNameInput"
+            name="name"
+            value={name}
+            onChange={(e) => onChange(e, id)}
+            isInvalid={isInvalid}
+          />
+        );
+      },
+    },
+    {
+      field: 'value',
+      name: i18n.translate('console.variablesPage.variablesTable.columns.valueHeader', {
+        defaultMessage: 'Value',
+      }),
+      render: (value, { id }) => (
+        <EuiFieldText
+          data-test-subj="variablesValueInput"
+          name="value"
+          onChange={(e) => onChange(e, id)}
+          value={value}
+        />
+      ),
+    },
+    {
+      field: 'id',
+      name: '',
+      width: '5%',
+      render: (id: string) => (
+        <EuiButtonIcon
+          iconType="trash"
+          aria-label="Delete"
+          color="danger"
+          onClick={() => deleteVariable(id)}
+          data-test-subj="variablesRemoveButton"
+        />
+      ),
+    },
+  ];
+
+  return (
+    <EuiFlyout onClose={props.onClose} ownFocus={false}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h2>
+            <FormattedMessage
+              id="console.variablesPage.pageTitle"
+              defaultMessage="Console Variables"
+            />
+          </h2>
+        </EuiTitle>
```

Comment:
I just realized this description is actually confusing, because you're not _defining_ the variables with the `${ ... }` format, you're consuming them. So maybe this text would make more sense:

> Define variables and use them in your requests in the form of <code>${variableName}</code>.

WDYT?

Reply:
Thanks for all of these suggestions! It's looking much better with these changes